### PR TITLE
support AbstractArrays

### PR DESCRIPTION
**Pull request by bjarthur**

### CODE REVIEW

File: src/HDF5.jl [hunk 1]
```diff
@@ -1614,7 +1614,7 @@ for (privatesym, fsym, ptype) in
         end
         # Scalar types
         ($fsym)(parent::$ptype, name::String, data::Union{T, AbstractArray{T}}, plists...) where {T<:ScalarOrString} =
-            ($privatesym)(parent, name, data, plists...)
+            ($privatesym)(parent, name, typeof(data)<:AbstractArray ? collect(data) : data, plists...)
```
Comment (musm) [inline src/HDF5.jl]: You can just use `data isa AbstractArray` or `isa(data,AbstractArray)` here

**Comment by bjarthur** [comment]
do you know why travis is called twice?  once labeled pr the other labelled push.  Gadfly does this as well.  i think it's not necessary and would like to turn it off there.

**Comment by musm** [comment]
> do you know why travis is called twice? once labeled pr the other labelled push. Gadfly does this as well. i think it's not necessary and would like to turn it off there.


Not sure about that

**Comment by bjarthur** [comment]
curious too that the tests pass on the linux nightly but not mac or windows.

**Comment by musm** [comment]
There is  a winrpm bug on 0.7 right now

**Comment by musm** [comment]
I feel like this should also  be mentioned in the readme.  It's a little concerning to do the wholesale conversion automatically. Perhaps the conservative approach where the user is forced to call `collect` is better/safer.

**Comment by musm** [comment]
anyone else have thoughts on this?

**Comment by bjarthur** [comment]
i could go either way.  it's more Julian i guess to force the user to call collect, and would be less maintenance for HDF5.jl.  on the other hand, it is convenient.

would need to revert https://github.com/JuliaIO/HDF5.jl/pull/464 and https://github.com/JuliaIO/HDF5.jl/pull/463 too.  is that as easy as pushing the "revert" button on github?  never done that.

could also consider revoking my push privileges.  i'm busy enough with Gadfly so am happy to give up that responsibility, which seemingly i overstepped here.

**Comment by musm** [comment]
I feel more comfortable, without wholesale conversion. I do however understand the convenience of the current approach. However, since HDF5 does not save type information I think it is reasonable to explicitly force the user to convert to an array, so they understand that HDF5 does not have the ability to retain type information, in which case they should use JLD/JLD2.

**Comment by musm** [comment]
> would need to revert #464 and #463 too. is that as easy as pushing the "revert" button on github? never done that.

should be that simple

**Comment by bjarthur** [comment]
i'm fine with that.  go for it!

**Comment by musm** [comment]
we can open up a further issue to discuss and obtain other opinions in the meantime

**Pull request merged**